### PR TITLE
Fix a bunch of TS types

### PR DESCRIPTION
**Pull request by LudvigHz**
- Fix TS types in server/
- Fix types in comments, images, user components

### CODE REVIEW

File: app/routes/events/utils.ts [hunk 1]
```diff
@@ -272,21 +274,24 @@ export const eventStatusTypes = [
...
   return (
     find(eventStatusTypes, {
       value: eventStatusType,
     }) || eventStatusTypes[0]
   );
 };
+
 export const getEventSemesterFromStartTime = (
-  startTime: moment
+  startTime: Moment
```
Comment (eikhr) [inline app/routes/events/utils.ts:287]:
```suggestion
  startTime: Dateish
```
It is used with `event.startTime` which is `Dateish`

File: app/routes/events/utils.ts [hunk 2]
```diff
@@ -10,10 +10,6 @@ import type {
   EventSemester,
 } from 'app/models';
 
-export const PHOTO_CONSENT_DOMAINS = {
-  WEBSITE: 'WEBSITE',
-  SOCIAL_MEDIA: 'SOCIAL_MEDIA',
-};
```
Comment (eikhr) [inline app/routes/events/utils.ts:16]: This is still imported and (attempting to be) used in `RegistrationMeta.tsx` and `RegistrationTables.tsx`, update them to the new enum.

**Review by eikhr — CHANGES_REQUESTED** [review]
Great work! 

I've just commented some nitpicking, and also you forgot to update to your new enum a couple of places causing a crash.

File: app/reducers/users.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export type UserEntity = {
   profilePicture: string;
   profilePicturePlaceholder?: string;
   emailListsEnabled?: boolean;
-  selectedTheme: string;
+  selectedTheme?: string;
   photoConsents?: Array<PhotoConsent>;
```
Comment (ivarnakken) [inline app/reducers/users.ts:22]: What's the point of keeping the `UserEntity` type when we have a `User` model in _app/models.ts_? Are they not the same? It seems like the only difference is `selectedTheme` and `emailListEnabled`.

File: app/reducers/comments.ts [hunk 4]
```diff
@@ -9,10 +9,11 @@ export type CommentEntity = {
   id: ID;
   parent?: ID;
   createdAt: string;
-  children: Array<Record<string, any>>;
+  children?: Array<Record<string, any>>;
```
Comment (ivarnakken) [inline app/reducers/comments.ts:12]: Could you type this better while you're at it :))

File: app/components/Layout/Flex.tsx [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import cx from 'classnames';
 import styles from './Flex.css';
-import type { ReactNode, ElementType } from 'react';
+import type { ReactNode, ElementType, HTMLAttributes } from 'react';
```
Comment (ivarnakken) [inline app/components/Layout/Flex.tsx:3]: I now realized that I'd prefer if our import sort put `react` imports at the top and style imports at the bottom. Just a thought.

File: app/components/Image/ProfilePicture.tsx [hunk 6]
```diff
@@ -1,16 +1,18 @@
 import type { UserEntity } from 'app/reducers/users';
 import CircularPicture from './CircularPicture';
+import type { ComponentProps } from 'react';
```
Comment (ivarnakken) [inline app/components/Image/ProfilePicture.tsx:3]: Is this not sorted wrongly? Why is it not caught by our linter?

**Review by ivarnakken — APPROVED** [review]
This is extremely satisfying to review! 😍

File: app/components/Layout/Flex.tsx (hunk 5 above)
Comment (ivarnakken) [reply app/components/Layout/Flex.tsx:3]: I now also realized this is a type import, so forget what I said about react, but the style import remains.

File: app/components/Image/ProfilePicture.tsx (hunk 6 above)
Comment (LudvigHz) [reply app/components/Image/ProfilePicture.tsx:3]: heh, our sorting rules are a bit _off_. Everything is sorted automatically, so this is how it wants it. 
I think the `app/` group overrides type imports :/ Should look at it, but not in this PR

Comment (ivarnakken) [reply app/components/Image/ProfilePicture.tsx:3]: Yes, definitely out of scope here.

**Comment by LudvigHz** [comment]
Heads up! I originally wanted to merge this, and do incremental type updates. But I touched on a lot of the same code in my other branch, so I decided to include those commits here as well. It will be a bit larger, but I think it's better considering some of the issues mentioned here might be fixed by those commits :)

File: app/reducers/users.ts (hunk 3 above)
Comment (LudvigHz) [reply app/reducers/users.ts:22]: Yeah, the whole double typing thing is ridiculous. But I'm trying to move more and more types into the `app/models`. The `***Entity` types seem to be more for creating new entities and typing the actions, so we'll keep them for now.

In some of the changes here, you can see I replace a bunch of the `Entity` types with better types from `app/models` and I think that's the way to go here.

File: app/reducers/comments.ts (hunk 4 above)
Comment (LudvigHz) [reply app/reducers/comments.ts:12]: I don't really want to :grimacing:.

Trying to focus on the TS errors, and typing the `any`s that are pretty easy. This one I don't really know what should be so I don't want to spend a lot of time on it.

File: app/reducers/meetings.ts [hunk 7]
```diff
@@ -8,19 +8,22 @@ export type MeetingEntity = {
   id: number;
   title: string;
   location: string;
-  startTime: moment$Moment;
-  endTime: moment$Moment;
+  startTime: Moment;
+  endTime: Moment;
```
Comment (ivarnakken) [inline app/reducers/meetings.ts:12]: Are these not `Dateish`?

File: app/models.ts [hunk 8]
```diff
@@ -233,16 +300,35 @@ export type CompanySemesterContactedStatus =
...
   | 'not_contacted';
+
+export type Meeting = {
+  id: ID;
+  createdBy: ID;
+  title: string;
+  location: string;
+  startTime: Moment;
+  endTime: Moment;
+  reportAuthor: number | null;
```
Comment (ivarnakken) [inline app/models.ts:311]: Shouldn't this rather be of type `ID`?

File: app/models.ts [hunk 9]
```diff
@@ -233,16 +300,35 @@ export type CompanySemesterContactedStatus =
...
   | 'contacted'
   | 'not_contacted';
+
+export type Meeting = {
+  id: ID;
+  createdBy: ID;
+  title: string;
+  location: string;
+  startTime: Moment;
+  endTime: Moment;
```
Comment (ivarnakken) [inline app/models.ts:310]: Same as comment above.

File: app/routes/events/components/RegistrationMeta.tsx [hunk 10]
```diff
@@ -58,7 +58,7 @@ const ConsentStatus = ({
   if (!useConsent) return null;
 
   if (photoConsents && allConsentsAnswered(photoConsents)) {
-    const { WEBSITE, SOCIAL_MEDIA } = PHOTO_CONSENT_DOMAINS;
+    const { WEBSITE, SOCIAL_MEDIA } = PhotoConsentDomain;
```
Comment (ivarnakken) [inline app/routes/events/components/RegistrationMeta.tsx:61]: Is it not bad practice to destruct an enum?

Comment (LudvigHz) [reply app/routes/events/components/RegistrationMeta.tsx:61]: Nah, it works just like an object, just with proper types. So it doesn't really matter

Comment (ivarnakken) [reply app/routes/events/components/RegistrationMeta.tsx:61]: I just felt like it kinda defeated its purpose, even thought it kinda doesn't ..

**Review by eikhr — APPROVED** [review]

**Pull request merged**